Doc/c-api/typeobj.rst

```diff
@@ -664,6 +664,10 @@ and :c:type:`PyType_Type` effectively act as defaults.)
    :c:func:`PyObject_GC_Del` if the instance was allocated using
    :c:func:`PyObject_GC_New` or :c:func:`PyObject_GC_NewVar`.
 
+   Finally, if the type is heap allocated (:const:`Py_TPFLAGS_HEAPTYPE` or
+   built with :c:func:`PyType_FromSpec`), the deallocator should decrement the
```

Comment:
I would remove the parenthesis here as it's already implied that a heap allocated type can only come from a managed class or a type created through `PyType_FromSpec`.

That being said, this might be obvious to me but not to any other readers so maybe we can do the following: Leave the reference to `:const:`Py_TPFLAGS_HEAPTYPE`. Then, modify `Py_TPFLAGS_HEAPTYPE` to include that this is set to any python class from a *.py file as well as those created through `PyType_FromSpec`, just like `Py_TPFLAGS_READY` specifies with `PyType_Ready`. Now from here you can click on the flag which will have a reference as to how types with this flag are created.

Reply:
I like that idea, I left the reference to `:const:Py_TPFLAGS_HEAPTYPE` and that mentions `PyType_FromSpec`. Regarding this part: 

>from type.__new__ file

Could you help me out with the wording? Not sure if it's the best place to mention it as this document deals entirely with the C API but it still might be useful.